Automattic/wp-calypso
# Add rewindBackupsRequestStatus selector and redux logic.

#### Changes proposed in this Pull Request

This is just an example PR - sort of a proof of concept.

Related to Slack convo:  p1643181075084200-slack-CDLH4C1UZ

This PR re-structures the redux logic to be able to query the request **_status_** of `wpcom/v2/sites/${siteId }/rewind/backups` that is triggered from the `requestRewindBackups( siteId)` action (or `<QueryRewindBackups/>` component).

- Adds logic to read new action types for `REWIND_BACKUPS_REQUEST_SUCCESS` and `REWIND_BACKUPS_REQUEST_FAILURE`;
- Adds a new `getRewindBackupsRequestStatus( state, siteId )` selector.

**Notes:**
This is just a draft and it could probably use some refactoring to make it cleaner/more readable. Additionally it is not using typescript.  If we decide to implement this, (or some version of this), we should probably convert it to typescript.

#### Testing instructions

1. Checkout this PR and `yarn start`.
2. Open http://calypso.localhost:3000.
3. Select a Jetpack site that has a Jetpack Backup subscription (or any plan that includes Backup).
4. Navigate to: Jetpack --> Backup - ( http://calypso.localhost:3000/backup/:SITE )
5. Open your developer console. Reload the page and observe the console.  You should see 3 different log entries as the page loads and as the status of the rewind backups request changes:
```
Rewind backup request status:  null
...
Rewind backup request status:  pending
...
Rewind backup request status:  success
...
```

Usage: 
Import the selector:
`import getRewindBackupsRequestStatus from 'calypso/state/selectors/get-rewind-backups-request-status';`

Then in your react component:

```
const siteId = useSelector( getSelectedSiteId );

const rewindBackupRequestStatus = useSelector( ( state ) =>
	getRewindBackupsRequestStatus( state, siteId )
);
```



Related to: p1643181075084200-slack-CDLH4C1UZ
